service/src/main/java/bio/terra/landingzone/library/landingzones/management/AzureResourceTypeUtils.java

```diff
@@ -13,4 +15,24 @@ private AzureResourceTypeUtils() {}
   public static final String AZURE_LOG_ANALYTICS_WORKSPACE_TYPE =
       "Microsoft.OperationalInsights/workspaces";
   public static final String AZURE_SOLUTIONS_TYPE = "Microsoft.OperationsManagement/solutions";
+
+  public static final String RESOURCE_ID_FORMAT =
+      "subscriptions/{guid}/resourceGroups/{resource-group-name}/{resource-provider-namespace}/{resource-type}/{resource-name}";
+
+  public static final String resourceTypeFromResourceId(String resourceID) {
+    if (StringUtils.isBlank(resourceID)) {
+      throw new IllegalArgumentException("resource id is blank");
+    }
+
+    String[] segments = StringUtils.split(StringUtils.strip(resourceID.trim(), "/"), "/");
+
+    // there should be 8 segments in a valid resource id.
+    if (segments.length != 8) {
+      throw new IllegalArgumentException(
+          "The resource id is not in the correct format. The format must be:" + RESOURCE_ID_FORMAT);
+    }
+
+    // the second to last segment is the resource provider
+    return String.format("%s/%s", segments[5], segments[6]);
```

Comment:
Maybe creating constants for the particular parts might increase readability. For instance, RESOURCE_TYPE_SEGMENT_INDEX=5